### PR TITLE
feat: bind UUID arrays for DuckDB appends

### PR DESCRIPTION
**Pull request by artie-hermes**
## What & Why
Binds NDJSON UUID-array elements to DuckDB UUID values before appending, preserving null array elements and returning indexed errors for malformed UUIDs.

## Verification
- `git diff --check`
- Focused append and conversion regression tests added.
- Not run locally: this environment has no Go toolchain; CI will run the Go suite.

## Rollback
`git revert 4dbc1fd`

Deploy before the dependent Transfer UUID-array PR.

---

> [!NOTE]
> **Medium Risk**
> Changes append-time type conversion for UUID arrays and removes explicit DOUBLE coercion, which could affect edge cases (e.g. string-encoded doubles) though tests still expect native float pass-through.
> 
> **Overview**
> **NDJSON append** now converts `UUID[]` columns by parsing each array element into `duckdb.UUID`, leaving **null** slots untouched and surfacing **indexed** conversion errors for bad UUID strings. Scalar `UUID` conversion is refactored through a shared `convertUUID` helper.
> 
> The dedicated **`DOUBLE`** branch in `ConvertValue` is removed; double columns now rely on the default pass-through path (same as other unhandled types).
> 
> Regression coverage adds **append** integration for UUID arrays (including null elements) and **unit** tests for valid/invalid UUID array conversion.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 76f6eab714b25508ea6971a7afd8614b1ce7cc4b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: internal/utils/utils.go [hunk 1]
```diff
@@ -155,17 +184,6 @@ func ConvertValue(value any, columnMetadata ColumnMetadata) (driver.Value, error
 			return nil, fmt.Errorf("failed to parse time %q for column %q (expected type %s)", s, columnMetadata.Name, metadataType)
 		}
 		return value, nil
-	case "DOUBLE":
```
Comment (Tang8330) [inline internal/utils/utils.go:158]: undoing changes from here: https://github.com/artie-labs/ducktape/pull/55

because it applied in transfer already.

Comment (carolkng) [reply internal/utils/utils.go:158]: 🙏

**Review by carolkng — APPROVED** [review]
